### DISCUSSION TITLE
issue: registered name bug

### DISCUSSION THREAD

**Discussion by goongoguma** [react-hook-form/react-hook-form#9242] [Q&A] [answered]
### Version Number

7.26.1

### Codesandbox/Expo snack

https://codesandbox.io/s/practical-shape-994jmc?file=/src/App.tsx

### Steps to reproduce

1. Assign a name such as 91.aabbbcccddd
2. Write value in textfield
3. Click Submit
4. Check console.log

### Expected behaviour

Expected Value when the form is submitted

```
{ "91.aabbbcccddd": "hello" }
```

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
{
  91: {
    aaabbbcccddd: "hello"
  }
}
```

### Code of Conduct

- [O] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@goongoguma : Field name with dot is not supported in react-hook-form as it breaks nested fields handling because nested field name can be a flat field name with dot (.), for example "parent.children.[0].firstName".

If the source data is your own data, I would suggest to change the data structure following convention of most database engines which does not allow dot (.) inside field name. Even if database engine support field name with dot (.), it comes with many limitation on indexing, select query or full text search ...

If the source data is external data, you might need to replace dot (.) with something like underscore "_" then replace field name when submitting to server-side or post-processing on server-side

**Reply by goongoguma** [reply]
Thanks for the answer. I have decided to use default input field instead.
